### PR TITLE
Remove the "yield" from attack modules

### PR DESCRIPTION
**Pull request by devl00p**
First pass of trying to remove the "yield" from attack modules

### CODE REVIEW

**Comment by devl00p** [comment]
All modules were tested and seems to behave normally. Ctrl+C feature works correctly along with count of network errors.

**Review by bretfourbe — COMMENTED** [review]
Interesting changes :)
Ensure to rebase on master though

File: wapitiCore/attack/mod_csp.py [hunk 1]
```diff
@@ -30,17 +32,33 @@ class mod_csp(Attack):
...
+        if self.finished:
+            return False
+
+        if request.method == "POST":
+            return False
+
+        return request.url == self.persister.get_root_url()
+
+    def attack(self, request: Request):
+        request_to_root = Request(request.url)
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/attack/mod_csp.py]: done

File: wapitiCore/attack/mod_nikto.py [hunk 2]
```diff
@@ -99,7 +100,10 @@ def update(self):
         except IOError:
             print(_("Error downloading nikto database."))
 
-    def attack(self):
+    def must_attack(self, request: Request):
+        return not self.finished
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/attack/mod_nikto.py:104]: done, I missed this one

File: wapitiCore/attack/mod_brute_login_form.py [hunk 3]
```diff
@@ -83,97 +82,97 @@ def test_credentials(self, login_form, username_index, password_index, username,
...
                 "invalid", "invalid"
             )
 
             if self.check_success_auth(failure_text):
-                # Ignore this case as it raise false positives
-                continue
+                # Ignore this case as it raises false positives
+                return
+        except RequestException:
+            return
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/attack/mod_brute_login_form.py]: done

File: wapitiCore/attack/mod_cookieflags.py [hunk 4]
```diff
@@ -36,9 +36,13 @@ def check_secure_flag(cookie: object):
...
-        url = self.persister.get_root_url()
-        request = Request(url)
+    def must_attack(self, request: Request):
+        if request.method == "POST":
+            return False
+
+        return request.url == self.persister.get_root_url()
+
+    def attack(self, request: Request):
         cookies = self.crawler.session_cookies
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/attack/mod_cookieflags.py]: done

File: wapitiCore/attack/mod_http_headers.py [hunk 5]
```diff
@@ -33,24 +36,40 @@ class mod_http_headers(Attack):
...
+        if self.finished:
+            return False
+
+        if request.method == "POST":
+            return False
+
+        return request.url == self.persister.get_root_url()
+
+    def attack(self, request: Request):
+        request_to_root = Request(request.url)
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/attack/mod_http_headers.py:59]: done

**Comment by devl00p** [comment]
> Interesting changes :)
> Ensure to rebase on master though

Should be ok, not exactly how I wanted it to be but code is pushed and squashed.

**Pull request closed without merging**